app-misc/lcd4linux/lcd4linux-0.11.0_pre20170527.ebuild

```diff
@@ -0,0 +1,195 @@
+IUSE="${IUSE} $(printf 'lcd_devices_%s ' ${IUSE_LCD_DEVICES[@]})"
+REQUIRED_USE="${REQUIRED_USE} || ( $(printf 'lcd_devices_%s ' ${IUSE_LCD_DEVICES[@]}) )"
+
+# Add supported plugins from 'IUSE_LCD4LINUX_PLUGINS' to 'IUSE' and 'REQUIRED_USE'
+IUSE="${IUSE} $(printf '%s ' ${IUSE_LCD4LINUX_PLUGINS[@]})"
+REQUIRED_USE="${REQUIRED_USE} || ( $(printf '%s ' ${IUSE_LCD4LINUX_PLUGINS[@]}) )"
+
+# Define dependencies for all drivers in 'IUSE_LCD_DEVICES'
+DEPEND_LCD_DEVICES="lcd_devices_astusb? ( virtual/libusb:0= )
+	lcd_devices_bwct? ( virtual/libusb:0= )
+	lcd_devices_curses? ( sys-libs/ncurses:0= )
+	lcd_devices_dpf? ( virtual/libusb:0= )
+	lcd_devices_g15? ( virtual/libusb:0= )
+	lcd_devices_glcd2usb? ( virtual/libusb:0= )
+	lcd_devices_irlcd? ( virtual/libusb:0= )
+	lcd_devices_lcd2usb? ( virtual/libusb:0= )
+	lcd_devices_ledmatrix? ( virtual/libusb:0= )
+	lcd_devices_luise? ( dev-libs/luise-bin
+		virtual/libusb:0= )
+	lcd_devices_matrixorbitalgx? ( virtual/libusb:0= )
+	lcd_devices_mdm166a? ( virtual/libusb:0= )
+	lcd_devices_picolcd? ( virtual/libusb:0= )
+	lcd_devices_picolcdgraphic? ( virtual/libusb:0= )
+	lcd_devices_png? ( media-libs/gd[png]
+		media-libs/libpng:0= )
+	lcd_devices_ppm? ( media-libs/gd )
+	lcd_devices_samsungspf? ( virtual/libusb:0= )
+	lcd_devices_serdisplib? ( dev-libs/serdisplib )
+	lcd_devices_shuttlevfd? ( virtual/libusb:0= )
+	lcd_devices_trefon? ( virtual/libusb:0= )
+	lcd_devices_ula200? ( dev-embedded/libftdi:1=
+		virtual/libusb:0= )
+	lcd_devices_usbhub? ( virtual/libusb:0= )
+	lcd_devices_usblcd? ( virtual/libusb:0= )
+	lcd_devices_vnc? ( net-libs/libvncserver )
+	lcd_devices_x11? ( x11-libs/libX11 )"
+
+# Define dependencies for all plugins in 'IUSE_LCD4LINUX_PLUGINS'
+DEPEND_LCD4LINUX_PLUGINS="asterisk? ( net-misc/asterisk )
+	dbus? ( sys-apps/dbus )
+	gps? ( dev-libs/nmeap )
+	hddtemp? ( app-admin/hddtemp )
+	iconv? ( virtual/libiconv )
+	mpd? ( media-libs/libmpd )
+	mpris_dbus? ( sys-apps/dbus )
+	mysql? ( || ( dev-db/mariadb-connector-c
+		dev-db/mysql-connector-c ) )
+	python? ( ${PYTHON_DEPS} )
+	wireless? ( || ( net-wireless/iw
+		net-wireless/wireless-tools ) )"
+
+RDEPEND="dmalloc? ( dev-libs/dmalloc )
+	${DEPEND_LCD_DEVICES}
+	${DEPEND_LCD4LINUX_PLUGINS}"
+
+DEPEND="${RDEPEND}"
+
+pkg_setup() {
+	if use python; then
+		python-single-r1_pkg_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	# Install new version of ax_python_devel.m4, to fix support for >=dev-lang/python-3.
+	# Patch also plugin_python.c, to compile with >=dev-lang/python-3.
+	# The compilation seems also not to link against >=dev-lang/python-3 correctly, so adding it to the libs.
+	# See https://lcd4linux.bulix.org/wiki/plugin_python for more information.
+	if use python; then
+		if ! use python_single_target_python2_7; then
+			cp "${FILESDIR}"/ax_python_devel.m4 ax_python_devel.m4 || die
+			eapply "${FILESDIR}"/${PN}-0.11.0-add_python3_support.patch
+			append-libs "-lpython${EPYTHON#python}m"
```

Comment:
Why? This looks like a serious bug in the build system. In any case, I'd say remove python3 support unless you can get it fixed properly upstream. Adding a lot of downstream hacks just to get py3 supported is not worth the effort.

Reply:
Okay, fine for me. I am anyway not using the python plugin. Will drop it completely, as I don't think, this will be fixed ever by upstream.